html/js/graph.js

```diff
@@ -0,0 +1,1044 @@
+    var databases = JSON.parse(window.localStorage.getItem(key_localstorage_databases));
+    if (databases === null) {
+        var data_score_databases;
+        $.ajax({
+            type: "GET",
+            url: URL_ROOT_API + ENDPOINT_SCORE_DBS,
+            async: false,
+            contentType: "application/json; charset=utf-8",
+            success: function (data) {
+                data_score_databases = filterDatabases(data);
+                cacheScoreDatabases(data_score_databases);
+            }
+        });
+
+        return data_score_databases;
+    } else {
+        return databases;
+    }
+}
+
+function getCachedCurrentDatabase() {
+    var key_localstorage_current_db = LOCAL_STORAGE_CURRENT_DATABASE;
+    const current_db = JSON.parse(window.localStorage.getItem(key_localstorage_current_db));
+
+    return current_db;
+}
+
+function filterDatabases(databases) {
+    const dbs_to_show = databases.stats.filter(function (db) {
+        // Mock for filtering only active dbs for everybody and all dbs for testing purposes
+        if (window.location.href.search("/test") >= 0 || db.isActive == true) {
+            return db;
+        }
+
+      return null;
+    });
+    databases.stats = dbs_to_show;
+
+    return databases;
+}
+
+function cacheScoreDatabases(databases) {
+    var key_localstorage_databases = LOCAL_STORAGE_SCORE_DATABASES;
+    window.localStorage.setItem(key_localstorage_databases, JSON.stringify(databases));
+}
+
+function cacheCurrentScoreDatabase(current_db) {
+    var key_localstorage_current_db = LOCAL_STORAGE_CURRENT_DATABASE;
+    if (current_db !== null && current_db !== undefined) {
+        window.localStorage.setItem(key_localstorage_current_db, JSON.stringify(current_db));
+    }
+}
+
+function changeScoreDb(ctrl) {
+  current_db_for_graph = getCachedScoreDatabases().stats[ctrl.selectedIndex];
+  cacheCurrentScoreDatabase(current_db_for_graph);
+}
+
+// **************
+// products_suggestion
+// **************
+
+/* Note:
+ [x, y]: x = number of product selected once products filtered and ordered
+ note: in order to access the circle drawn in the graph, use suggested_products[x].num_circle
+ y = DOM-image of selected product
+ */
+var suggested_products = [];
+
+/*
+ [x, y]: x = number of product selected once products filtered and ordered
+ note: in order to access the circle drawn in the graph, use suggested_products[x].num_circle
+ y = DOM-image of selected product
+ */
+var client_current_selection = [-1, undefined];
+
+function cleanup_suggestions() {
+
+    /* clear previous suggestions if any */
+    $(ID_PRODUCTS_SUGGESTION).empty();
+    $(ID_NB_SUGGESTIONS).empty();
+    $(ID_NB_SUGGESTIONS).append(0);
+    $(ID_PRODUCTS_SUGGESTION).attr("height", `String(6 + $(window).innerHeight() / 3)` + "px");
+    $(ID_PRODUCTS_SUGGESTION).append("<ul></ul>");
+    $(ID_MENU_SELECTION).attr("height", `String($(window).innerHeight() / 5)` + "px");
```

Comment:
This doesn't seem to work (same for height of `ID_PRODUCTS_SUGGESTION`) for the p. In my browser, this results into `<div id="menu_selection" height="String($(window).innerHeight() / 5)px">`, which is not a valid height.

Reply:
Fixed it up.